crates/biome_lsp/src/handlers/analysis.rs

```diff
@@ -149,11 +149,25 @@ pub(crate) fn code_actions(
             if has_quick_fix && action.suggestion.applicability == Applicability::MaybeIncorrect {
                 return None;
             }
+            // Filter out source.organizeImports.biome action when organize imports is not supported.
             if action.category.matches("source.organizeImports.biome")
                 && !file_features.supports_organize_imports()
             {
                 return None;
             }
+            // Filter out quickfix.biome action when lint is not supported.
+            // But keep JavaScript parse rules because they're not configurable
+            // and should be considered as a parse action
+            if action.category.matches("quickfix.biome")
+                && action.rule_name.as_ref().map_or(true, |(_, name)| {
+                    name != "noDuplicatePrivateClassMembers"
+                        && name != "noInitializerWithDefinite"
+                        && name != "noSuperWithoutExtends"
+                })
```

Comment:
To be honest, I don't know if we want to do this...

Reply:
I don't think we should

Reply:
Okay, I dropped them in d4f6acc